### ISSUE TITLE
Support key-value annotations on wrap

### ISSUE THREAD

**Issue by abhinav** [bracesdev/errtrace#36]
Based on @prashantv's suggestions.
In the future, we should consider adding support for `errtrace.Wrap` to include key-value pairs as part of wrapping.
These values can be shown next to the function call in the trace, e.g.

```
example.com/main.getUser name="foo"
    path/to/file:42
```

The API for this is undecided at this time.

**Comment by StevenACoffman** [comment]
[cockroachdb/errors](https://github.com/cockroachdb/errors) has a ton of useful functionality... but is a very heavyweight and complicated library.

In #106 I added an interface so that the main value proposition of this library is composable with external custom error types.

I hope that allows this particular project to maintain a tighter focus.

**Comment by abhinav** [comment]
@StevenACoffman Agreed! We're hoping to keep this library lightweight and focused, and not turn it into a general "error utils" library.

Whether this feature goes into this library is still up for discussion, and it would have to be lightweight and composable (similarly to #106) if it did. An argument can be made that this is a logging-specific concern and it should go in a logerr package; nothing about this in set in stone at this time.

**Comment by gregwebs** [comment]
I [have a package for attaching slog values and stack traces to errors](https://pkg.go.dev/github.com/gregwebs/errors/slogerr#Wraps).
I think they probably do need to be done either in the same function call or an additional method call.

I was considering the possibility of switching over to using errtrace for my errors packages.
However, errtrace doesn't allow for creating new wrapping functions, only using its.
If you look in the link I sent, you will see a function [WrapsSkip](https://pkg.go.dev/github.com/gregwebs/errors/slogerr#WrapsSkip), where "skip" refers to skipping function calls in the stack trace. I don't see this kind of ability in errtrace other than to try to use its internal/pc package which does expose these abilities.

**Comment by prashantv** [comment]
@gregwebs I merged a change to master which should allow other wrappers, see https://github.com/bracesdev/errtrace/pull/108

It hasn't been released yet, but you can try that out by pointing to master.

**Comment by gregwebs** [comment]
@prashantv thanks! That should do it.
